app/models/dialog_field_sorted_item.rb

```diff
@@ -96,6 +96,9 @@ def sort_data(data_to_sort)
   end
 
   def determine_selected_default_value
+    if dynamic? && force_multi_value && !default_value.kind_of?(Array)
```

Comment:
@romanblanco doesn't it make more sense to put this *after*` the `use_first_value_as_default unless default_value_included?(@raw_values)`?

If you `Array.wrap` first, `default_value_included?` will never return true - doesn't this break it for some cases? (Like, `[nil]` is probably not what you want.)

Also, if `use_first_value_as_default` does get called, it won't wrap the default value in the array (until you re-run this method twice).

Or is this intended?

Reply:
@himdel I've tried, and there's another problem while ordering a dialog containing dialog containing dynamic dropdown list after the changes you've suggested.

The workflow fails on 500 error because of calling `collect` operation on `nil`.

If this doesn't block the PR from merging, I'd say it'll be better to fix this in a following PR, as the situation is a bit confusing to me, and we want to merge this soon.

Reply:
Aaah, because https://github.com/ManageIQ/manageiq/pull/16978 overrides it to do some JSON magic, without making sure `default_value` is a JSON... this is really broken :(

Looks like `default_value` can be anything from an array to a JSON-encoded array :(

I don't know. The fix looks buggy without it, but the bug looks larger than the fix :).

Cc @eclarizio

Reply:
Either way, I guess you need to know *why* you get that error @romanblanco.

Either `use_first_value_as_default` is broken and should never be called in your situation.
Or it's supposed to be called and needs to be fixed too.

Reply:
@himdel I think the line:
```
use_first_value_as_default unless default_value_included?(@raw_values)
```
is fine to stay as it is. The values list will always end up being an array of arrays, so there's no problem with `use_first_value_as_default`. The issue is in `default_value_included?`, because the `default_value` can be either a single value, or multiple values, but string encoded (this is where the JSON.parse comes in). To fix this bug, I think that just needs to be modified to handle when the default value is not set, when it's multiple values, and when it's a single value.

Reply:
OK, no complaints for me then :)

Reply:
@eclarizio You may want to adjust your review, still shows up as `x`

Reply:
@eclarizio if this can be merged in the next few hours, we probably can get this into today's blocker-only build. If not, can you let me know so I can pass the info along.